playground/streaming/benchmark.py

```diff
@@ -0,0 +1,551 @@
+                        f"[Transcriber: {transcriber_name}] Transcriber queue is empty, stopping transcription..."
+                    )
+                    send_audio.cancel()
+                    break
+            if not args.transcriber_use_mic:
+                pbar.update(pbar.total - pbar.n)
+            transcriber.terminate()
+
+
+async def main():
+    result_file_path = os.path.join(args.results_dir, args.results_file)
+    if not args.just_graphs:
+        if args.agents:
+            await run_agents()
+        if args.transcribers:
+            await run_transcribers()
+        if args.synthesizers:
+            await run_synthesizers()
+
+        trace_results = span_exporter.get_finished_spans()
+        final_spans = defaultdict(list)
+        for span in trace_results:
+            duration_ns = span.end_time - span.start_time
+            duration_s = duration_ns / 1e9
+            final_spans[span.name].append(duration_s)
+
+        scope_metrics = reader.get_metrics_data().resource_metrics[0].scope_metrics
+        if len(scope_metrics) > 0:
+            metric_results = scope_metrics[0].metrics
+            metric_results = {
+                metric.name: metric.data.data_points[0] for metric in metric_results
+            }
+            final_metrics = {}
+            for metric_name, raw_metric in metric_results.items():
+                if re.match(r"transcriber.*\.min_latency", metric_name):
+                    final_metrics[metric_name] = raw_metric.min
+                if re.match(r"transcriber.*\.max_latency", metric_name):
+                    final_metrics[metric_name] = raw_metric.max
+                if re.match(r"transcriber.*\.avg_latency", metric_name):
+                    transcriber_str = metric_name.split(".")[1]
+                    final_metrics[metric_name] = (
+                        raw_metric.sum
+                        / metric_results[f"transcriber.{transcriber_str}.duration"].sum
+                    )
+                if re.match(r"agent.*\.total_characters", metric_name):
+                    agent_str = metric_name.split(".", 1)[1].rsplit(".", 1)[0]
+                    final_metrics[
+                        f"agent.{agent_str}.characters_per_second"
+                    ] = raw_metric.value / sum(
+                        final_spans[f"agent.{agent_str}.generate_total"]
+                    )
+
+        final_spans = {k: sum(v) / len(v) for k, v in final_spans.items()}
+        if len(scope_metrics) > 0:
+            final_results = {**final_spans, **final_metrics}
+        else:
+            final_results = final_spans
+        print(json.dumps(final_results, indent=4))
+        if args.results_file:
+            with open(result_file_path, "w") as f:
+                json.dump(final_results, f, indent=4)
+    else:
+        with open(result_file_path, "r") as f:
+            final_results = json.load(f)
+
+    if args.create_graphs or args.just_graphs:
```

Comment:
nit: I would define helper methods for creating graphs and just call it here, cleans up the method a bit